### PR TITLE
use ES modules

### PR DESCRIPTION
**Pull request by davidchambers**
The time has come! :D

### CODE REVIEW

File: .config [hunk 1]
```diff
@@ -1,6 +1,6 @@
 repo-owner = davidchambers
 repo-name = doctest
 author-name = David Chambers <dc@davidchambers.me>
-source-files = bin/* lib/*.js lib/*.mjs
+source-files = bin/doctest lib/*.js lib/*.mjs
```
Comment (davidchambers) [inline .config:4]: This change is necessary because we are adding __bin/package.json__.

File: .eslintrc.json [hunk 2]
```diff
@@ -1,9 +1,10 @@
 {
   "root": true,
   "extends": ["./node_modules/sanctuary-style/eslint-es6.json"],
+  "parserOptions": {"ecmaVersion": 2020, "sourceType": "module"},
   "env": {"node": true},
-  "globals": {
-    "Promise": "readonly"
+  "rules": {
+    "comma-dangle": ["error", {"arrays": "always-multiline", "objects": "always-multiline", "imports": "always-multiline", "exports": "always-multiline", "functions": "never"}]
```
Comment (davidchambers) [inline .eslintrc.json:7]: This is necessary while we await a sanctuary-scripts release that includes sanctuary-js/sanctuary-style#35.

File: .travis.yml [hunk 3]
```diff
@@ -2,10 +2,10 @@ os: linux
 dist: focal
 language: node_js
 node_js:
-  - "10.2.0"
+  - "12.22.0"
```
Comment (davidchambers) [inline .travis.yml:5]: [Node v12.22.0][1] was the first version to support ES modules without requiring a command line flag.


[1]: https://nodejs.org/en/blog/release/v12.22.0/

File: bin/doctest [hunk 4]
```diff
@@ -11,10 +11,9 @@ const flags = idx >= 0 && idx < args.length - 1;
 require ('child_process')
 .spawn (
   process.execPath,
-  ['--experimental-modules',
```
Comment (davidchambers) [inline bin/doctest:14]: This is no longer necessary. :)

File: package.json [hunk 5]
```diff
@@ -12,14 +12,15 @@
...
+  "type": "module",
+  "main": "./lib/doctest.js",
   "bin": "./bin/doctest",
   "repository": {
     "type": "git",
     "url": "git://github.com/davidchambers/doctest.git"
   },
   "engines": {
-    "node": ">=10.2.0"
+    "node": ">=12.22.0"
```
Comment (davidchambers) [inline package.json:23]: [Node v12.22.0][1] was the first version to support ES modules without requiring a command line flag.


[1]: https://nodejs.org/en/blog/release/v12.22.0/

File: test/index.js [hunk 6]
```diff
@@ -49,7 +48,7 @@ const testCommand = (command, expected) => {
     try {
       return {
         status: 0,
-        stdout: execSync (`${command} --nodejs --no-warnings`, options),
+        stdout: execSync (command, options),
```
Comment (davidchambers) [inline test/index.js:51]: We no longer need to suppress warnings as ES modules are no longer experimental. :)

**Review by Avaq — APPROVED** [review]

File: lib/doctest.js [hunk 7]
```diff
@@ -507,60 +520,60 @@ const log = results => {
...
+                ((options.type === 'coffee' ||
+                  options.type == null && ext === '.coffee') ?
+                 rewriteCoffee :
+                 compose (compose (toModule (options.module)))
+                         (rewriteJs ('script')))
+                (options.module === 'commonjs' ?
+                 commonjsEval (path) :
+                 functionEval);
+  }
+};
```
Comment (davidchambers) [inline lib/doctest.js:579]: Aldwin, I greatly reduced the amount of repetition. I hope you like the changes. :)

**Review by Avaq — APPROVED** [review]

**Pull request merged**